src/Features/CSharp/Portable/CodeRefactorings/UseRecursivePatterns/UseRecursivePatternsCodeRefactoringProvider.cs

```diff
@@ -117,14 +117,29 @@ private static bool IsFixableNode(SyntaxNode node)
                 };
             }
 
-            if (TryGetCommonReceiver(leftReceiver, rightReceiver, model) is var (commonReceiver, leftNames, rightNames))
+            if (TryGetCommonReceiver(leftReceiver, rightReceiver, leftTarget, rightTarget, model) is var (commonReceiver, leftNames, rightNames))
             {
                 return root =>
                 {
-                    var leftSubpattern = CreateSubpattern(leftNames, CreatePattern(leftReceiver, leftTarget, leftFlipped));
-                    var rightSubpattern = CreateSubpattern(rightNames, CreatePattern(rightReceiver, rightTarget, rightFlipped));
-                    var replacement = IsPatternExpression(commonReceiver, RecursivePattern(leftSubpattern, rightSubpattern));
-                    return root.ReplaceNode(logicalAnd, AdjustBinaryExpressionOperands(logicalAnd, replacement));
+                    if (leftNames.Length == 0)
+                    {
+                        var rightSubpattern = CreateSubpattern(rightNames, CreatePattern(rightReceiver, rightTarget, rightFlipped));
+                        var replacement = IsPatternExpression(commonReceiver, RecursivePattern(rightSubpattern));
+                        return root.ReplaceNode(logicalAnd, AdjustBinaryExpressionOperands(logicalAnd, replacement));
+                    }
+                    else if (rightNames.Length == 0)
+                    {
+                        var leftSubpattern = CreateSubpattern(leftNames, CreatePattern(leftReceiver, leftTarget, leftFlipped));
+                        var replacement = IsPatternExpression(commonReceiver, RecursivePattern(leftSubpattern));
+                        return root.ReplaceNode(logicalAnd, AdjustBinaryExpressionOperands(logicalAnd, replacement));
+                    }
```

Comment:
please doc these cases.  a good way to do this is to show the sort of code it would trigger for, and what it is replacing that code with.

Reply:
Added some comments explaining the logic